src/inmanta/protocol/methods_v2.py

```diff
@@ -405,3 +405,27 @@ def get_resource_actions(
     :raises BadRequest: When the supplied parameters are not valid.
 
     """
+
+
+@typedmethod(
+    path="/resource/<resource_id>/deploy/start",
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def resource_deploy_start(
+    tid: uuid.UUID,
+    resource_id: str,
+    action_id: uuid.UUID,
+) -> Dict[str, ResourceState]:
```

Comment:
Why `str` and not `ResourceVersionId`?

Reply:
Because protocol cannot handle this. Protocol expects a type annotation of type class, but `ResourceVersionId` has type function because if was created using `typing.NewType()`.

Reply:
Right, I've just encountered this while working on my changes. I've added a fix on my branch so I'll try to update this type annotation on my branch after you've merged it in.